bazel/scripts/run_integ_tests.sh

```diff
@@ -0,0 +1,239 @@
+}
+
+create_test_targets() {
+    if [[ "${TARGET_PATH}" == *":"* ]];
+    then
+        echo "Single target specified - running test:"
+        categorize_test "${TARGET_PATH}"
+    else
+        echo "Multiple targets specified - running tests:"
+        create_extended_test_targets
+        # TODO: other tests
+    fi
+    if [[ "${#EXTENDED_TEST_TARGETS[@]}" -eq 0 ]]; # TODO other tests
+    then
+        echo "ERROR: No test found."
+        help
+        exit 1
+    fi
+    for TARGET in "${EXTENDED_TEST_TARGETS[@]}"
+    do
+        echo "${TARGET}"
+    done
+}
+
+create_extended_test_targets() {
+    mapfile -t EXTENDED_TEST_TARGETS < <(bazel query "attr(tags, extended_test, kind(py_test, //lte/gateway/python/integ_tests/s1aptests/...))")
+}
+
+list_all_tests() {
+    echo "All integration tests:"
+    list_extended_tests
+    # TODO: other tests
+    exit 0
+}
+
+list_extended_tests() {
+    echo "Extended tests:"
+    create_extended_test_targets
+    for TARGET in "${EXTENDED_TEST_TARGETS[@]}"
+    do
+        echo "${TARGET}"
+    done
+}
+
+list_traffic_server_tests() {
+    echo "Tests that require the traffic server:"
+    bazel query "attr(tags, traffic_server_test, kind(py_test, //lte/gateway/python/integ_tests/s1aptests/...))"
+}
+
+setup_extended_tests() {
+    echo "Setting up the environment for the extended tests."
+    echo "Building..."
+    bazel build "//lte/gateway/python/integ_tests/s1aptests:test_modify_mme_config_for_sanity" --define=on_magma_test=1
+    echo "Executing..."
+    sudo "${MAGMA_ROOT}/bazel-bin/lte/gateway/python/integ_tests/s1aptests/test_modify_mme_config_for_sanity"
+    echo "Setup finished successfully."
+}
+
+teardown_extended_tests() {
+    echo "Cleaning up the environment after the extended tests."
+    echo "Building..."
+    bazel build "//lte/gateway/python/integ_tests/s1aptests:test_restore_mme_config_after_sanity" --define=on_magma_test=1
+    echo "Executing..."
+    sudo "${MAGMA_ROOT}/bazel-bin/lte/gateway/python/integ_tests/s1aptests/test_restore_mme_config_after_sanity"
+    echo "Cleanup finished successfully."
+}
+
+run_test() {
+    local TARGET=$1
+    local TARGET_PATH=${TARGET%:*}
+    local SHORT_TARGET=${TARGET#*:}
+    (
+        set -x
+        echo "BUILDING TEST: ${TARGET}"
+        bazel build "${TARGET}" --define=on_magma_test=1
+        echo "RUNNING TEST: ${TARGET}"
+        sudo "${MAGMA_ROOT}/bazel-bin/${TARGET_PATH}/${SHORT_TARGET}"
+    )
+}
+
+print_summary() {
+    local NUM_SUCCESS=$1
+    local TOTAL_TESTS=$2
+    echo "SUMMARY: ${NUM_SUCCESS}/${TOTAL_TESTS} tests were successful."
+    for TARGET in "${!TEST_RESULTS[@]}"
+    do
+        echo "  ${TARGET}: ${TEST_RESULTS[${TARGET}]}"
+    done
+}
+
+###############################################################################
+# SCRIPT SECTION
+###############################################################################
+
+TARGET_PATH="${1:-}"
+
+if [[ "${TARGET_PATH}" == "--help" ]];
```

Comment:
Have we thought about using a separate tool or other mechanism for parsing the command line arguments? Right now the order is also relevant, e.g. when adding the `--skip-setup-teardown-extended` option (line 202ff) it does not work if this is the first argument and then a test is passed on afterwards. 

But it might add too much complexity. 

Some ideas can be found here:
- https://github.com/matejak/argbash
- https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash

Reply:
Good idea. Fixed.